packages/core-v2-sdk/src/datasources/MultiTerm/MultiTermContractDataSource.ts

```diff
@@ -119,4 +121,72 @@ export class MultiTermContractDataSource
     const supply = await this.call("totalSupply", [termId]);
     return fromBn(supply, await this.getDecimals());
   }
+
+  /**
+   * Wraps the lock function in the Term contract, allows caller lto mint fixed and variable positions in a term.
+   * @async
+   * @param {Signer} signer - Ethers signer object.
+   * @param {string[]} assetIds -  The array of PT, YT and Unlocked share identifiers.
+   * @param {string[]} assetAmounts - The amount of each input PT, YT and Unlocked share to use
+   * @param {number} termId - The term id (expiry).
+   * @param {string} amount - Amount of underlying tokens to use to mint.
+   * @param {string} ptDestination - Address to receive principal tokens.
+   * @param {string} ytDestination - Address to receive yield tokens.
+   * @param {string} hasPreFunding- Have any funds already been sent to the contract, not commonly used for EOAs.
+   * @return {Promise<MintResponse>}
+   */
+  async lock(
+    signer: Signer,
+    termId: number,
+    assetIds: string[],
+    assetAmounts: string[],
+    amount: string,
+    ptDestination: string,
+    ytDestination: string,
+    ytBeginDate: number,
+    hasPreFunding: boolean,
+  ): Promise<MintResponse> {
+    if (assetAmounts.length !== assetIds.length) {
+      throw new Error(
+        "Error MultiTermDataSource.Lock(): assetIds and assetAmounts must be the same length.",
```

Comment:
We can remove the "Error" part. It'll get added automatically if this error is uncaught or by the `console.error` method.